src/panels/config/js/trigger/geo_location.js

```diff
@@ -0,0 +1,117 @@
+import { h, Component } from "preact";
+import "@polymer/paper-radio-button/paper-radio-button";
+import "@polymer/paper-radio-group/paper-radio-group";
+import "../../../../components/entity/ha-entity-picker";
+
+import { onChangeEvent } from "../../../../common/preact/event";
+
+const SOURCES = [
```

Comment:
Not sure if this is a good idea. I rather have people enter the constant themselves. Can we make a list based on the platforms that are loaded? (available as `hass.config.components`)

Reply:
Re making use of `hass.config.components`: At this point, each geolocation platform defines a `SOURCE` constant that happens to be the same as the platform name. So, I could go through the loaded components, and simply pick each `geo_location.` one. Not sure though if this will work forever, i.e. if a geolocation platform ever defines a `SOURCE` different to its own name.

Another option could be to implement that in the backend, i.e. each loaded geolocation platform registers its `SOURCE` which in turn will be made available to the frontend as for example `hass.config.geo_location.sources`.

Or, we could go with a simple input field for now, and then add some form of verification or input helper later?

Reply:
OK, I transformed the source dropdown into a text input field.